api/src/domain/collection.rs

```diff
@@ -41,52 +43,52 @@ pub struct CollectionRequest {
     pub body: Option<RequestBody>,
 }
 
-#[derive(Debug, Deserialize, Serialize)]
+#[derive(Debug, Deserialize, Serialize, PartialEq, Clone)]
 pub struct CollectionUrl {
     pub raw: String,
     pub host: Option<Vec<String>>,
     pub path: Option<Vec<String>>,
 }
 
-#[derive(Debug, Deserialize, Serialize)]
+#[derive(Debug, Deserialize, Serialize, PartialEq, Clone)]
 pub struct RequestBody {
     pub mode: String,
-    pub raw: String,
+    pub raw: Option<String>,
```

Comment:
This was causing some imports to fail since not all requests will have a raw body set